### PR TITLE
Fix #267

### LINKED ISSUE #267: DDNet Crashes on Ubuntu 15.04
When I was trying to connect to a server,it told me like this

```
[15-08-01 21:13:43][client]: loading map, map=Mountain_Shadow wanted crc=682c1a94
[15-08-01 21:13:43][datafile]: loading. filename='maps/Mountain_Shadow.map'
[15-08-01 21:13:43][datafile]: could not open 'maps/Mountain_Shadow.map'
[15-08-01 21:13:43][datafile]: loading. filename='downloadedmaps/Mountain_Shadow_682c1a94.map'
[15-08-01 21:13:43][datafile]: could not open 'downloadedmaps/Mountain_Shadow_682c1a94.map'
[15-08-01 21:13:43][client/network]: starting to download map to 'downloadedmaps/Mountain_Shadow_682c1a94.map'
[15-08-01 21:13:43][fetcher]: Thread started...
[15-08-01 21:13:43][fetcher]: Task got http://maps.ddnet.tw/Mountain_Shadow_682c1a94.map:downloadedmaps/Mountain_Shadow_682c1a94.map
[15-08-01 21:13:43][fetcher]: Downloading downloadedmaps/Mountain_Shadow_682c1a94.map
[15-08-01 21:13:45][fetcher]: Task failed. libcurl error: name lookup timed out
```

and after I restarted the client, the log in terminal showed this and the client crashed.

```
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  4 (X_DestroyWindow)
  Resource id in failed request:  0x440000f
  Serial number of failed request:  139
  Current serial number in output stream:  141
```

Should I provide any other information?

### PR DESCRIPTION
**Pull request by godwhoa**
Fixes crashes caused by dns timeout.

### CODE REVIEW

**Comment by def-** [comment]
Looks ok to me, @Learath2 ?

**Comment by heinrich5991** [comment]
Reference: https://curl.haxx.se/libcurl/c/CURLOPT_NOSIGNAL.html

Looks like this stops the ability of curl to time out on DNS requests.

**Comment by godwhoa** [comment]
[Explanation](http://stackoverflow.com/questions/21887264/why-libcurl-needs-curlopt-nosignal-option-and-what-are-side-effects-when-it-is/21902693#21902693) and [testing demo](https://youtu.be/R2X3jxUBNtc).

File: src/engine/client/fetcher.cpp [hunk 1]
```diff
@@ -143,7 +143,8 @@ void CFetcher::FetchFile(CFetchTask *pTask)
 	curl_easy_setopt(m_pHandle, CURLOPT_NOPROGRESS, 0);
 	curl_easy_setopt(m_pHandle, CURLOPT_PROGRESSDATA, pTask);
 	curl_easy_setopt(m_pHandle, CURLOPT_PROGRESSFUNCTION, &CFetcher::ProgressCallback);
-
+	curl_easy_setopt(m_pHandle, CURLOPT_NOSIGNAL, 1L);
+
```
Comment (heinrich5991) [inline src/engine/client/fetcher.cpp]: Really minor nit, you don't have to fix this: Whitespace on lines without other stuff can generally be removed, the diff would be shorter if you hadn't inserted this tab.

Comment (godwhoa) [inline src/engine/client/fetcher.cpp]: Oops. code editor inserted it automatically will keep in mind next time.

**Comment by heinrich5991** [comment]
This is clearly something we want according to the documentation.

**Comment by heinrich5991** [comment]
Thanks for the pull request!

**Comment by Learath2** [comment]
Lets hope this was it.

**Pull request merged**